File: HaL5/principles-of-programming-languages/lab6/exe5.hs
filter' :: (a -> Bool) -> [a] -> [a]
filter' _ [] = []
filter' p lst = [x | x <- lst, p x]

main = do
	print(filter' (\x -> x > 2) [1,2,3,4,5])
	print(filter' (\x -> x < 6.5) [7.9, 1.2, 12.44, 6.4, 3.33])

File: HaL5/advanced-programming/lab4/review/exe5.py
#!/usr/bin/env python

def binary_to_decimal(n):
    return int(n, 2)

if __name__ == "__main__":
    lst = [i for i in input().split(",")]
    res = list()
    for i in lst:
        if binary_to_decimal(i) % 5 == 0:
            res.append(i)
    print(res)
